test/built-ins/RegExp/lookbehind.js

```diff
@@ -0,0 +1,169 @@
+         c, and performs the following steps:
+        a. Let d be a Continuation that always returns its State argument as a successful MatchResult.
+        b. Call m(x, d) and let r be its result.
+        c. If r is failure, return failure.
+        d. Let y be r's State.
+        e. Let cap be y's captures List.
+        f. Let xe be x's endIndex.
+        g. Let z be the State (xe, cap).
+        h. Call c(z) and return its result.
+features: [regexp-lookbehind]
+includes: [compareArray.js]
+---*/
+
+// Simple fixed-length matches.
+assert.compareArray(["a"], "a".match(/^.(?<=a)/));
+assert.sameValue(null, "b".match(/^.(?<=a)/));
+assert.compareArray(["foo"], "foo1".match(/^f..(?<=.oo)/));
+assert.compareArray(["foo"], "foo2".match(/^f\w\w(?<=\woo)/));
+assert.sameValue(null, "boo".match(/^f\w\w(?<=\woo)/));
+assert.sameValue(null, "fao".match(/^f\w\w(?<=\woo)/));
+assert.sameValue(null, "foa".match(/^f\w\w(?<=\woo)/));
+assert.compareArray(["def"], "abcdef".match(/(?<=abc)\w\w\w/));
+assert.compareArray(["def"], "abcdef".match(/(?<=a.c)\w\w\w/));
+assert.compareArray(["def"], "abcdef".match(/(?<=a\wc)\w\w\w/));
+assert.compareArray(["cde"], "abcdef".match(/(?<=a[a-z])\w\w\w/));
+assert.compareArray(["def"], "abcdef".match(/(?<=a[a-z][a-z])\w\w\w/));
+assert.compareArray(["def"], "abcdef".match(/(?<=a[a-z]{2})\w\w\w/));
+assert.compareArray(["bcd"], "abcdef".match(/(?<=a{1})\w\w\w/));
+assert.compareArray(["cde"], "abcdef".match(/(?<=a{1}b{1})\w\w\w/));
+assert.compareArray(["def"], "abcdef".match(/(?<=a{1}[a-z]{2})\w\w\w/));
+
+// Variable-length matches.
+assert.compareArray(["def"], "abcdef".match(/(?<=[a|b|c]*)[^a|b|c]{3}/));
+assert.compareArray(["def"], "abcdef".match(/(?<=\w*)[^a|b|c]{3}/));
+
+// Start of line matches.
+assert.compareArray(["def"], "abcdef".match(/(?<=^abc)def/));
+assert.compareArray(["def"], "abcdef".match(/(?<=^[a-c]{3})def/));
+assert.compareArray(["def"], "xyz\nabcdef".match(/(?<=^[a-c]{3})def/m));
+assert.compareArray(["ab", "cd", "efg"], "ab\ncd\nefg".match(/(?<=^)\w+/gm));
+assert.compareArray(["ab", "cd", "efg"], "ab\ncd\nefg".match(/\w+(?<=$)/gm));
+assert.compareArray(["ab", "cd", "efg"], "ab\ncd\nefg".match(/(?<=^)\w+(?<=$)/gm));
+assert.sameValue(null, "abcdef".match(/(?<=^[^a-c]{3})def/));
+assert.sameValue(null, "foooo".match(/"^foooo(?<=^o+)$/));
+assert.sameValue(null, "foooo".match(/"^foooo(?<=^o*)$/));
+assert.compareArray(["foo"], "foo".match(/^foo(?<=^fo+)$/));
+assert.compareArray(["foooo"], "foooo".match(/^foooo(?<=^fo*)/));
+assert.compareArray(["foo", "f"], "foo".match(/^(f)oo(?<=^\1o+)$/));
+assert.compareArray(["foo", "f"], "foo".match(/^(f)oo(?<=^\1o+)$/i));
+assert.compareArray(["foo\u1234", "f"], "foo\u1234".match(/^(f)oo(?<=^\1o+).$/i));
+assert.compareArray(["def"], "abcdefdef".match(/(?<=^\w+)def/));
+assert.compareArray(["def", "def"], "abcdefdef".match(/(?<=^\w+)def/g));
+
+// Word boundary matches.
```

Comment:
Would splitting up the tests help more than the comments in the test file helps? The tests in a particular file are all getting at the same bit of spec text.